### PR TITLE
Githubactions2

### PR DESCRIPTION
**Pull request by JusticeCarl**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-13564

### Change description ###

Final merge to incorporate Tim and Mohana's suggestions

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# finops-reporting-incremental-export
+
+Github actions run on a cron schedule to run scripts that interogate Cost Management APIs and upload to a Storage Account.
```
Comment (timja) [inline README.md:3]:
```suggestion
Github actions run on a cron schedule to run scripts that interrogate Cost Management APIs and upload to a Storage Account.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- Balance by period runs on the 1st of every month for the previous months data
+- Balance Summary export runs on the 1st of every month for the previous months data
+- Cost management export currently only uploads the relevant export to the portal and then it needs to be started manually in the portal in cost management. This is due to needing admin rights to run the post section of the process.
```
Comment (timja) [inline README.md:7]:
```suggestion
- Cost management export currently only uploads the relevant export to the portal and then it needs to be started manually in the portal in cost management. This is due to needing admin rights to start the export via API but reader permissions being enough in the portal.
```

File: scripts/reservationcharges.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+IFS="|"
+data_source="reservation_charges"
+
+# Set date command based on OS
+platform=$(uname)
+if [[ $platform == "Darwin" ]]; then
+    date_command=$(which gdate)
+elif [[ $platform == "Linux" ]]; then
+    date_command=$(which date)
+fi
```
Comment (timja) [inline scripts/reservationcharges.sh:14]: could this be added to all the scripts not just this one please? you can probably extract it to one file and then source it

Comment (JusticeCarl) [reply scripts/reservationcharges.sh]: added a functions.sh which is sourced by all scripts and moved this logic into it.

File: scripts/functions.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+#  Set date command based on OS
+date_cmd () {
```
Comment (timja) [inline scripts/functions.sh:3]: how is this working?

it's invoked at `date_command` but I can't see anything calling `date_cmd`

Comment (JusticeCarl) [reply scripts/functions.sh]: yes you're correct. removed the uncalled function, changed functions.sh to common.sh. changed all references in dependant scripts to source common.sh and call $date_command. tested all scripts call the date command as expected.

**Pull request merged**